### DISCUSSION TITLE
v2.6.0

### DISCUSSION THREAD

**Discussion by daanx** [koka-lang/koka#409] [General]
If you use VSCode try out the new language server support, and install the compiler through the extension. Leave feedback here!

<img align="right" width="200" src="https://github.com/koka-lang/koka/raw/master/doc/system-logos.png">
After installing, read the <a href="https://koka-lang.github.io/koka/doc/book.html">getting started guide</a>.

## Linux (x64) and macOS (x64, arm64 (M1/M2))

Tested on Ubuntu 18, 20, Debian 10, macOS Catalina (x64), and Ventura (M1), and should run on most Linux distributions.
From a command prompt, run:
```
curl -sSL https://github.com/koka-lang/koka/releases/download/v2.6.0/install.sh | sh
```
This also installs syntax highlighting for the VS Code and Atom editors. After install, run `koka` to verify if koka installed correctly.

* For most installations this will ask for root access in order to install to `/usr/local/bin`. For more control, you can pass a different prefix. For example, installing to `~/.local` instead:  
  `curl -sSL https://github.com/koka-lang/koka/releases/download/v2.6.0/install.sh | sh -s -- --prefix=~/.local`

* To uninstall a version, use the `--uninstall` option:  
  `curl -sSL https://github.com/koka-lang/koka/releases/download/v2.6.0/install.sh | sh -s -- --uninstall`

## Windows 10/11 (x64)

Open a `cmd` prompt and download and run the installer:
```
curl -sSL -o %tmp%\install-koka.bat https://github.com/koka-lang/koka/releases/download/v2.6.0/install.bat && %tmp%\install-koka.bat
```
This will also prompt to install the [`clang`](https://llvm.org/builds) compiler if needed, and install syntax highlighting for the VS Code and Atom editors. After install, run `koka` to verify if koka installed correctly. 

* On Windows, the default install is to the user profile at `%APPDATA%\local`. You can change the installation directory using `--prefix`. For example:  
  `curl -sSL -o %tmp%\install-koka.bat https://github.com/koka-lang/koka/releases/download/v2.6.0/install.bat && %tmp%\install-koka.bat --prefix=c:\programs\local`

* To uninstall a version, use the `--uninstall` option:  
  `curl -sSL -o %tmp%\install-koka.bat https://github.com/koka-lang/koka/releases/download/v2.6.0/install.bat && %tmp%\install-koka.bat --uninstall`

## Other platforms

You need to build [from source](https://github.com/koka-lang/koka#build-from-source), however Koka has few dependencies and should build from source without problems on most common platforms.

<hr /><em>This discussion was created from the release <a href='https://github.com/koka-lang/koka/releases/tag/'>v2.6.0</a>.</em>

**Comment by TimWhiting** [comment]
The main update in this release is Language Server support. Currently we have focused on the VSCode extension for LS support. However, since the language server protocol is editor agnostic, it should be straightforward to support other editors if someone would like to contribute support.

Also, FYI you can now directly install / update the Koka compiler from the VSCode extension, and anytime the extension is updated it will check for a new release if the one on your computer has a different version than the latest release. If you decline the install initially there is a command you can trigger via VSCode's command menu to install at a later point.
